lib/improver/blending/weights.py

```diff
@@ -238,12 +242,575 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+                then model_configuration may appropriate.
+            config_dict (dict):
+                Dictionary containing the configuration information, namely
+                an initial set of weights and information regarding the
+                points along the specified coordinate at which the weights are
+                valid. An example dictionary is shown below.
+
+        Keyword Args:
+            weights_coord_name (str):
+                Name of the string within the configuration dictionary used
+                to specify the weights. This name is also used as the name
+                of the cube output from this plugin.
+
+        Dictionary of format:
+        ::
+            {
+                "uk_det": {
+                    "forecast_period": [7, 12],
+                    "weights": [1, 0],
+                    "units": "hours"
+                }
+                "uk_ens": {
+                    "forecast_period": [7, 12, 48, 54]
+                    "weights": [0, 1, 1, 0]
+                    "units": "hours"
+                }
+            }
+
+        """
+        self.config_dict = config_dict
+        self.weighting_coord_name = weighting_coord_name
+        self.config_coord_name = config_coord_name
+        self.weights_coord_name = weights_coord_name
+        self._check_config_dict()
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        msg = ("<ChooseWeightsLinearFromDict config_dict = {}, "
+               "weighting_coord_name = {}, config_coord_name = {}, "
+               "weights_coord_name = {}>".format(
+                   str(self.config_dict), self.weighting_coord_name,
+                   self.config_coord_name, self.weights_coord_name))
+        return msg
+
+    def _check_config_dict(self):
+        """Check whether the items within the configuration dictionary
+        are of matching lengths.
+
+        Raises:
+            ValueError: If items within the configuration dictionary are
+                not of matching lengths.
+        """
+        # Check all keys
+        for key in self.config_dict.keys():
+            weighting_len = (
+                len(self.config_dict[key][self.weighting_coord_name]))
+            weights_len = (
+                len(self.config_dict[key][self.weights_coord_name]))
+            if weighting_len != weights_len:
+                msg = ("{} is {}, {} is {}."
+                       "These items in the configuration dictionary "
+                       "have different lengths i.e. {} != {}".format(
+                           self.weighting_coord_name,
+                           self.config_dict[key][self.weighting_coord_name],
+                           self.weights_coord_name,
+                           self.config_dict[key][self.weights_coord_name],
+                           weighting_len, weights_len))
+                raise ValueError(msg)
+
+    def _arrange_interpolation_inputs(self, cube):
```

Comment:
This might be better as `_generate_inputs_from_dict`, also to distinguish it from the function in the `ChooseWeightsLinearFromCube` plugin (see below).